### ISSUE TITLE
Dependency not detected, build fails

### ISSUE THREAD

**Issue by dfclark** [Carthage/Carthage#1317]
I'm running carthage 0.16.2 on OS X 10.11.4 and am seeing this behavior. 
My cart file looks like this:

```
Dons-MBP:Project$ cat cartfile
github "CocoaLumberjack/CocoaLumberJack" ~> 2.0
github "Alamofire/Alamofire" ~> 3.3
github "SwiftyJSON/SwiftyJSON" ~> 2.3
git "git@gitserver.foobar.com:repositories/HMLoggerServicesFramework.git" "master"
```

But `carthage update --platform iOS` attempts to build HMLoggerServicesFramework before SwiftyJSON and the dependency on SwiftyJSON in HMLoggerServices causes this to fail.

The cartfile.resolved looks like this:

```
Dons-MBP:Project$ cat cartfile.resolved
github "Alamofire/Alamofire" "3.3.1"
github "CocoaLumberjack/CocoaLumberJack" "2.3.0"
git "git@gitserver.foobar.com:repositories/HMLoggerServicesFramework.git" "6ca53078005e7853f962f78e1ea90d3a59f4470b"
github "SwiftyJSON/SwiftyJSON" "2.3.3"
```

It tried the trick of editing cartfile.resolved, to this:

```
Dons-MBP:Project$ cat cartfile.resolved
github "SwiftyJSON/SwiftyJSON" "2.3.3"
github "Alamofire/Alamofire" "3.3.1"
github "CocoaLumberjack/CocoaLumberJack" "2.3.0"
git "git@gitserver.foobar.com:repositories/HMLoggerServicesFramework.git" "6ca53078005e7853f962f78e1ea90d3a59f4470b"
```

and then running the command `carthage build --platform iOS` but that made no difference, the build ordering stayed the same, and failed:

```
Dons-MBP:Project$ carthage build --platform iOS
*** xcodebuild output can be found in /var/folders/46/vq47kpz909jfv8j4vml8c63h0000gp/T/carthage-xcodebuild.p6ZV8p.log
*** Building scheme "Alamofire iOS" in Alamofire.xcworkspace
*** Building scheme "CocoaLumberjackSwift-iOS" in Lumberjack.xcworkspace
*** Building scheme "CocoaLumberjack" in Lumberjack.xcworkspace
*** Building scheme "CocoaLumberjackSwift" in Lumberjack.xcworkspace
*** Building scheme "CocoaLumberjack-iOS" in Lumberjack.xcworkspace
*** Building scheme "HMLoggerServices" in HMLoggerServices.xcworkspace
** BUILD FAILED **

The following build commands failed:
CompileSwift normal arm64 /Users/dfclark/Dev/HoboHH/Project/Carthage/Checkouts/HMLoggerServicesFramework/HMLoggerServices/WebServiceManager.swift
CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
/Users/dfclark/Dev/HoboHH/Project/Carthage/Checkouts/HMLoggerServicesFramework/HMLoggerServices/WebServiceManager.swift:16:8: error: no such module 'SwiftyJSON'
A shell task (/usr/bin/xcrun xcodebuild -workspace /Users/dfclark/Dev/HoboHH/Project/Carthage/Checkouts/HMLoggerServicesFramework/HMLoggerServices.xcworkspace -scheme HMLoggerServices -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
** BUILD FAILED **

The following build commands failed:
CompileSwift normal arm64 /Users/dfclark/Dev/HoboHH/Project/Carthage/Checkouts/HMLoggerServicesFramework/HMLoggerServices/WebServiceManager.swift
CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
```

**Comment by mdiep** [comment]
What does the `Cartfile` for `HMLoggerServicesFramework` look like? Is it located in the root directory of that repository?

**Comment by dfclark** [comment]
The Cartfile in the HMLoggerServices top-level folder (a peer of the Xcode project file) contains:
github "Alamofire/Alamofire" ~> 3.3
github "SwiftyJSON/SwiftyJSON" ~> 2.3

**Comment by mdiep** [comment]
Is `6ca53078005e7853f962f78e1ea90d3a59f4470b` the most recent revision of `HMLoggerServicesFramework`?

**Comment by dfclark** [comment]
Yes

**Comment by mdiep** [comment]
> The Cartfile in the HMLoggerServices top-level folder (a peer of the Xcode project file) contains:

Is that the root directory of the repository?

What's the output of `git show 6ca53078005e7853f962f78e1ea90d3a59f4470b:Cartfile <path-to-HMLoggerServices-repository>`?

**Comment by ikesyo** [comment]
Closing this due to inactivity. Feel free to reopen if you have any updates on this.

**Comment by ajaykoppisetty** [comment]
Same error. Deleted carthage folder and retried again. No success :(

Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project /Users/user1/Documents/Bitbucket/operator-ios/project.xcodeproj -scheme core-operator -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES build (launched in /Users/user1/Documents/Bitbucket/operator-ios)

** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64 /Users/user1/Documents/Bitbucket/core-operator-ios/project/Controllers/Inspections/InspectionsVC.swift
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)

Xcode build was successful. When tried to build with carthage these errors are showing up.

**Comment by mdiep** [comment]
Please open a new issue if you're looking for help.

**Comment by ikesyo** [comment]
Looks like @ajaykoppisetty's issue was resolved: #2226.
